pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -66,14 +66,19 @@
     private final ConcurrentOpenHashSet<PositionImpl> pendingAcks;
```

Comment:
Right now, we with these changes we are not considering batch-message usecase.
To include batch-message usecase:
A. broker needs to increase ```unackedMessages``` counter with number of batch messages are being sent. 
B. when ack has been received : broker needs to reduce ```unackedMessages``` counter with number of message present into received batch-message.

In order to achieve (B) we need to store : number-of-batch-messages in a given message at ```pendingAcks``` set. I have made this change into separate [commit](https://github.com/rdhabalia/pulsar/commit/2af7c186f9298105a4d10a82e21b023f8fd5b4d6) at [Consumer](https://github.com/rdhabalia/pulsar/blob/2af7c186f9298105a4d10a82e21b023f8fd5b4d6/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java#L66).
Should we include batch-message usecase for this feature Or broker should treat entire-batch-message as one message only? I think storing number of messages in a ```pendingAcks``` is not expensive so, we can support batch-message usecase.